temporal-serviceclient/src/main/java/io/temporal/serviceclient/ServiceStubsOptions.java

```diff
@@ -418,7 +418,7 @@ public String toString() {
   public static class Builder<T extends Builder<T>> {
     private ManagedChannel channel;
     private SslContext sslContext;
-    private boolean enableHttps;
+    private Boolean enableHttps;
```

Comment:
Here's a contrived situation:

* I create service stubs options with no API key and having set nothing about TLS whatsoever, thereby making `enableHttps` default to `false` on the service stub options
* I want to create new service stub options build from the existing options, so now `enableHttps` is `false` even though I never set it
* I do a `addApiKey` on this new service stub options I have never set any TLS setting on, but it still won't default to TLS because with the current code, it's impossible to create a builder from existing options and retain default TLS behavior

I don't mind if we break and/or don't-support this scenario, just wanted to mention it

Reply:
If we don't consider changing the return type of `ServiceStubOptions.getEnableHttps()` to `Boolean` instead of `boolean` to be too big of a compat break, I agree that is the better/clearer approach

Reply:
I didn't figure it'd be that common, but meh. This works for me, but what should be the return value of `getEnableHttps` be if `enableHttps` is `null`? Would the logic be `return (this.enableHttps != null && this.enableHttps) || (enableHttps == null && this.apiKeyProvided && this.sslContext == null && this.channel == null)`? I also assume users may never want to differentiate whether it's explicitly set or not? Maybe a different field name/getter and `getEnableHttps` is that more advanced logic but the getter still exists for the wrapper?